src/coreclr/tools/SuperFileCheck/Program.cs

```diff
@@ -0,0 +1,623 @@
+                }
+
+                if (IsArgumentCSharp(args[0]))
+                {
+                    if (args.Length == 1 || !IsArgumentCSharpFile(args[1]))
+                    {
+                        PrintErrorExpectedCSharpFile();
+                        return 1;
+                    }
+
+                    var checkFilePath = args[1];
+                    var checkFileNameNoExt = Path.GetFileNameWithoutExtension(checkFilePath);
+
+                    var hasInputFile = args.Any(x => x.Equals(CommandLineInputFile));
+                    if (!hasInputFile)
+                    {
+                        PrintErrorNoInputFileFound();
+                        return 1;
+                    }
+
+                    var checkPrefixes = DetermineCheckPrefixes(args);
+                    try
+                    {
+                        var methodDeclInfos = FindMethodsByFile(checkFilePath, checkPrefixes);
+
+                        if (!CheckDuplicateMethodNames(methodDeclInfos))
+                        {
+                            return 1;
+                        }
+
+                        if (!CheckMethodsHaveNoInlining(methodDeclInfos))
+                        {
+                            return 1;
+                        }
+
+                        if (methodDeclInfos.Length > 0)
+                        {
+                            var didSucceed = true;
+
+                            var tasks = new Task<FileCheckResult>[methodDeclInfos.Length];
+
+                            // Remove the first 'csharp' argument so we can pass the rest of the args
+                            // to LLVM FileCheck.
+                            var argsToCopy = args.AsSpan(1).ToArray();
+
+                            for (int i = 0; i < methodDeclInfos.Length; i++)
+                            {
+                                var index = i;
+                                var tmpFileName = $"__tmp{index}_{checkFileNameNoExt}.cs";
+                                var tmpDirName = Path.GetDirectoryName(checkFilePath);
+                                string tmpFilePath;
+                                if (String.IsNullOrWhiteSpace(tmpDirName))
+                                {
+                                    tmpFilePath = tmpFileName;
+                                }
+                                else
+                                {
+                                    tmpFilePath = Path.Combine(tmpDirName, tmpFileName);
+                                }
+                                tasks[i] = Task.Run(() => RunSuperFileCheckAsync(methodDeclInfos[index], argsToCopy.ToArray(), checkPrefixes, tmpFilePath));
+                            }
+
+                            await Task.WhenAll(tasks);
+
+                            foreach (var x in tasks)
+                            {
+                                if (x.Result.ExitCode != 0)
+                                    didSucceed = false;
```

Comment:
[nit] - You don't use the no-brace form anywhere else

Reply:
I'll put in the braces for consistency.